### PR TITLE
[Link Action] New style plugin

### LINKED ISSUE #3835: Reimplement Link Action in vanilla JS
The [Link Action](https://github.com/nasa/openmct/blob/master/platform/entanglement/src/actions/LinkAction.js) is currently implemented in AngularJS, and should be re-implemented in vanilla JavaScript. It should not make use of any legacy API, and should instead use our [Composition](https://github.com/nasa/openmct/blob/master/src/api/composition/CompositionAPI.js) and [Mutation ](https://github.com/nasa/openmct/blob/master/src/api/objects/ObjectAPI.js#L365) APIs.

### PR DESCRIPTION
**Pull request by jvigliotta**
Removed old link action legacy code and added new link action style plugin.

closes #3835 

### Author Checklist
Check | Passed
-- | --
Changes address original issue? | Y
Unit tests included and/or updated with changes? | Y
Command line build passes? | Y
Changes have been smoke-tested? | Y
Testing instructions included? | Y

### CODE REVIEW

File: src/plugins/linkAction/pluginSpec.js [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+describe("The Link Action plugin", () => {
+
+    let openmct;
+    let linkAction;
+    let childObject;
+    let parentObject;
+    let anotherParentObject;
+    const ORIGINAL_PARENT_ID = 'original-parent-object';
+    const LINK_ACITON_KEY = 'link';
```
Comment (shefalijoshi) [inline src/plugins/linkAction/pluginSpec.js:38]: Oh yeah!! `LINK_ACITON_KEY` :P

File: src/plugins/linkAction/LinkAction.js [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+                }
+            ]
+        };
+    }
+
+    validate(objectToLink) {
+        return (parentObject) => {
+            let parentCandidateKeystring = this.openmct.objects.makeKeyString(parentObject.getId());
+            let objectToLinkKeystring = this.openmct.objects.makeKeyString(objectToLink.identifier);
+            let sameObjectOrChildAlready = parentCandidateKeystring === objectToLinkKeystring
```
Comment (shefalijoshi) [inline src/plugins/linkAction/LinkAction.js:91]: Use `openmct.objects.areIdsEqual `here instead

File: src/plugins/linkAction/LinkAction.js [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+            if (sameObjectOrChildAlready) {
+                return false;
+            }
+
+            if (parentObject.getModel().locked) {
+                return false;
+            }
+
+            // can contain
+            return this.openmct.composition.checkPolicy(
```
Comment (shefalijoshi) [inline src/plugins/linkAction/LinkAction.js:104]: Can we get rid of this legacy code and use a new Composition policy instead?
See `ConditionSetCompositionPolicy`

File: src/plugins/linkAction/LinkAction.js [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+
+    async convertFromLegacy(legacyDomainObject) {
+        let objectContext = legacyDomainObject.getCapability('context');
+        let domainObject = await this.openmct.objects.get(objectContext.domainObject.id);
+
+        return domainObject;
+    }
+
+    linkInNewParent(child, newParent) {
+        let compositionCollection = this.openmct.composition.get(newParent);
```
Comment (shefalijoshi) [inline src/plugins/linkAction/LinkAction.js:62]: Hmm. Why not `newParent.composition.add(child)` ?

**Review by shefalijoshi — CHANGES_REQUESTED** [review]
Looks really good, suggesting some changes.

**Comment by jvigliotta** [comment]
Closing this as it's addressed in #3816

**Comment by jvigliotta** [comment]
handled in PR #3816

**Pull request closed without merging**